cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -4129,6 +4130,29 @@ data ScriptWitnessIndex =
    | ScriptWitnessIndexWithdrawal !Word
   deriving (Eq, Ord, Show)
 
+instance ToJSON ScriptWitnessIndex where
+  toJSON = \case
+    ScriptWitnessIndexTxIn n ->
```

Comment:
Manual instances are cumbersome in maintenance and error-prone. Maybe `genericToJSON` would be better here instead? For example:
```haskell
data Foo = Foo !Text | Bar !Text | Baz !Text deriving (Eq, Ord, Show, Generic)
genericToJSON (defaultOptions{sumEncoding=TaggedObject{tagFieldName="kind", contentsFieldName="value"}}) (Bar "barbar")
```
produces
```json
{"kind":"Bar","value":"barbar"}
```

Reply:
I'm not certain that buys enough to justify the cost for a number of reasons.

We need to do this for `LogFormatting` and have no generic facility for that type class.  If we have to manually type it out in one instance we may as well do it for both.

I use my editor's multi-cursor support and it takes very little time for something this regular.

My previous experience with `aeson` generic generation was that the compile time was quadratic and with a large enough type that actually became a problem.

Reply:
My main concern here is just an extra mental burden coming from the requirement to make proper changes in multiple places - what if you accidentally omit one place with your multi-cursor support? 😄

>My previous experience with aeson generic generation was that the compile time was quadratic and with a large enough type that actually became a problem.

Fair point, and I agree that it matters here. I'm not insisting on using `Generic` here, but it would be better to have more safety either by using automatic generation of instances, or testing them in property tests.

PS. I had to roll-back prod deployment in the past once, because of faulty, manually written `FromJSON` instance, which was undetected for over a year, so I guess I'm over-sensitive on this topic. 😄